Ensure Mentors completion displays 100% in the Dashboard

curt-mitch opened the pull request:
## Description
Mentors have the ability to complete a course project outside of the course platform. Therefore course completion values for mentors should not be calculated using the project completion rates.

Resolves https://github.com/OpenMined/openmined/issues/122

## Affected Dependencies
N/A

## How has this been tested?
manual testing in dev environment

## Checklist
- [X] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

curt-mitch committed:
Do not require project completion values for mentors

curt-mitch committed:
Check mentor status when displaying complete percentage on course overview

tcp committed:
Merge branch 'dev' into mentor-completion-percent-bug

tcp commented on apps/courses/src/components/CourseProgressCard.tsx:
```diff
@@ -36,11 +36,17 @@ export default ({ content, ...props }) => {
 
   const stats = getCourseProgress(progress, lessons, project?.parts);
-  const percentComplete =
+  let mentorState = props.ms;
```
Here, `mentorState` could be a prop instead of `ms` — obtained via destructuring the props. Also, one small suggestion is to improve the variable name. Maybe `isMentor` or `userIsMentor` makes more sense in this context than `mentorState` and `ms`.

tcp replied:
`const` instead of `let` as it is never reassigned

tcp commented on apps/courses/src/routes/courses/_helpers.ts:
```diff
@@ -343,3 +344,25 @@ export const useCoursePermissionGate = (user, lessons, page, params) => {
   return { page, user, lessons, params };
 };
+
+export const userIsMentor = async (user: firebase.User) => {
```
`react-hooks/rules-of-hooks` will complain about using hooks (useFirestore/useState/useEffect) inside this function due to the function being named `userIsMentor`. 

Having it as `useUserIsMentor` is... strange, in my opinion. As an improvement, we could have a hook that fetches the fs user. I've created issue #147 for this.

tcp replied:
Does not have to be async.

tcp commented on apps/courses/src/routes/courses/_helpers.ts:
```diff
@@ -343,3 +344,25 @@ export const useCoursePermissionGate = (user, lessons, page, params) => {
+    if (user && user.uid && !dbUser) fetchUser();
+  }, [user, dbUser, dbUserRef]);
+
+  const isMentor =
```
The only missing piece here is that we should verify if the current course is included in `mentorable_courses`

tcp commented on apps/courses/src/routes/courses/overview/index.tsx:
```diff
@@ -217,11 +218,19 @@ export default ({ course, page, progress }: CoursePagesProp) => {
     100;
 
+  // don't require project completion values for mentors
+  if (isMentor) {
```
Cool! Sounds better than `mentorState`!

tcp reviewed (commented):
Solid contribution @curt-mitch! I have created a new issue to tackle an improvement extracted from this PR (#147). I'll make a small change and merge. Thank you!

tcp committed:
refactor: apply changes after code review

Removed `ms` in favor of `isMentor` prop in CourseProgressCard, resolved
react-hooks/rules-of-hooks error in userIsMentor (renamed to
useIsMentor).

Made a slight modification to the `useIsMentor` hook -- now, the hooks
checks if the user is a mentor for a given course. A course must be
passed to the hook.
